### PR TITLE
Modify search term

### PR DESCRIPTION
**Pull request by TJakubek**
Simple proposal to match part of the search term instead of just the begining or end when searching for tokens. 
A similar example is the search in pancakeswap. 

As most of the wrapped tokens have 'wan' preceding them, the user has to write these 3 letters each time before any meaningful search occurs. Partial search allows to filter away the unnecessary tokens faster. 
A great example is just writing 'v' and wanVibe will be selectable, as the letter 'v' is only available in this one token. 

I also separated a function into a reusable one, though I'm not sure if it is really necessary as whitespace characters  aren't enterable into the search bar, and even if they are then there just won't be any results. 

I also considered adding highlighting so the filtered results would show where the match was found but 
a) it would alter too much of your code for little benefit, especially since the token list isn't too long,
b) other dex's manage without it.

To sum up, the only real change is in line 31, the rest can be removed if it is not in accordance with your style.

### CODE REVIEW

**Comment by lolieatapple** [comment]
looks good.

**Comment by TJakubek** [comment]
I reverted the comments and the function isolation since I didn't get any feedback and so not sure if its desired. 
Feel free to merge at earliest convenience :)

**Comment by TJakubek** [comment]
After consult with Mr FennecFox I have readded the isolated function
